src/headless/converse-chat.js

```diff
@@ -1110,11 +1110,29 @@ converse.plugins.add('converse-chat', {
                     return;
                 }
                 if (utils.isNewMessage(message) && this.isHidden()) {
+                    this.setFirstUnreadMsgId(message);
                     this.save({'num_unread': this.get('num_unread') + 1});
                     _converse.incrementMsgCounter();
                 }
             },
 
+            /**
+             * Sets the msgid of the first unread realtime message in a ChatBox.
+             * @param {_converse.Message} message
+             */
+            setFirstUnreadMsgId (message) {
+                if (this.get('num_unread') == 0) {
+                    const first_unread_id = this.get('first_unread_id');
+
+                    if (first_unread_id) {
+                      const msg = this.messages.get(first_unread_id);
+                      if (msg) msg.set("first_unread", false);
+                    }
+                    message.set("first_unread", true);
+                    this.set({'first_unread_id': message.get('id')});
```

Comment:
Instead of setting values on both the message and the chatbox, you can instead only set `first_unread_id` on the chatbox and then compare the message id with `first_unread_id` inside the template (or assign that to a variable and pass it in to the template).

That would simplify the shrink the code here.

I also think it's desirable to use `.save` instead of `.set` so that the value gets persisted across page loads.

Reply:
@jcbrand At this point, I need to ask fro a favor from you to finish this PR on my behalf. I have run out of time to complete this. I have started a holiday and will be away for a while. By the time I get back, this PR would have expired and I might have to start all over again.

As you have indicated with an emoji reaction like you like it in principle and have a clear idea of how you want it implemented, could please kindly merge it and finish off the implementation. It would really help avoid having to get my head around this PR in addition to all the other things I would have to face when I get back. Thanks.

Reply:
>Instead of setting values on both the message and the chatbox, you can instead only set first_unread_id on the chatbox 

Suggestions on how I would trigger onChanged  in converse-message-view.js without including first_unread in the list of properties to be monitored would be very much appreciated.

>I also think it's desirable to use .save instead of .set so that the value gets persisted across page loads.

ok.

Reply:
> Suggestions on how I would trigger onChanged in converse-message-view.js without including first_unread in the list of properties to be monitored would be very much appreciated.

Once we render messages with lit-element, we can simply listen for `change` on the chatbox because rendering will be efficient (if nothing changes in a message's markup the DOM doesn't update).

Unfortunately we're not there yet and currently all messages will be re-rendered every time :(

Sorry, I didn't consider the performance implications of having to re-render all messages when `first_unread_id` changes on the chatbox.

So it's good like this for now.

Please just change the entry the changelog (since there's no jumping to anything) and I'll merge.